appserver/topology/graph.go

```diff
@@ -0,0 +1,124 @@
+package topology
+
+import (
+	"encoding/json"
+)
+
+// Graph contains the groupds, edges and nodes of the graph.
+type Graph struct {
+	Nodes  []Node  `json:"nodes,omitempty" protobuf:"bytes,1,opt,name=nodes"`
+	Edges  []Edge  `json:"edges,name=edges"`
+	Groups []Group `json:"groups,name=groups"`
+}
+
+// Node is the id and name of a node.
+type Node struct {
+	ID   string `json:"id,name=id"`
+	Name string `json:"name,name=name"`
+}
+
+// Edge is the source and target of a node.
+type Edge struct {
+	Source string `json:"source,name=source"`
+	Target string `json:"target,name=target"`
+}
+
+// Group is a group of nodes.
+type Group struct {
+	ID    string   `json:"id,name=id"`
+	Name  string   `json:"name,name=name"`
+	Nodes []string `json:"nodes,name=nodes"`
+}
+
+// Resource of a node.
+type Resource struct {
+	Metadata string `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Status   string `json:"status,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Name     string `json:"name,name=name"`
+	Kind     string `json:"kind,name=kind"`
+}
+
+// NodeData is the node data.
+type NodeData struct {
+	Type      string     `json:"type,name=type"`
+	ID        string     `json:"id,name=id"`
+	Resources []Resource `json:"resource,omitempty" protobuf:"bytes,1,opt,name=resource"`
+	Data      Data       `json:"data,name=data"`
```

Comment:
`Name` of the workload should be a high level entry here.

Reply:
done here: 06080cacb02f9e974d0523a36fce04a29a42b50b !